Signal/src/ViewControllers/ContactShareViewHelper.swift

```diff
@@ -0,0 +1,289 @@
+//
+//  Copyright (c) 2018 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+import SignalServiceKit
+import ContactsUI
+import MessageUI
+
+@objc
+public protocol ContactShareViewHelperDelegate: class {
+    func didCreateOrEditContact()
+}
+
+@objc
+public class ContactShareViewHelper: NSObject, CNContactViewControllerDelegate {
+
+    weak var delegate: ContactShareViewHelperDelegate?
+
+    let contactShare: ContactShareViewModel
+    let contactsManager: OWSContactsManager
+    weak var fromViewController: UIViewController?
+
+    public required init(contactShare: ContactShareViewModel, contactsManager: OWSContactsManager, fromViewController: UIViewController, delegate: ContactShareViewHelperDelegate) {
+        SwiftAssertIsOnMainThread(#function)
+
+        self.contactShare = contactShare
+        self.contactsManager = contactsManager
+        self.fromViewController = fromViewController
+        self.delegate = delegate
+
+        super.init()
+    }
+
+    // MARK: Actions
+
+    @objc
+    public func sendMessageToContact() {
+        Logger.info("\(logTag) \(#function)")
+
+        presentThreadAndPeform(action: .compose)
+    }
+
+    @objc
+    public func audioCallToContact() {
+        Logger.info("\(logTag) \(#function)")
+
+        presentThreadAndPeform(action: .audioCall)
+    }
+
+    @objc
+    public func videoCallToContact() {
+        Logger.info("\(logTag) \(#function)")
+
+        presentThreadAndPeform(action: .videoCall)
+    }
+
+    private func presentThreadAndPeform(action: ConversationViewAction) {
+        // TODO: We're taking the first Signal account id. We might
+        // want to let the user select if there's more than one.
+        let phoneNumbers = contactShare.systemContactsWithSignalAccountPhoneNumbers(contactsManager)
```

Comment:
The one change is that we're now leveraging the new "phone number convenience accessors" in `OWSContact` and `ContactShareViewModel`.